packages/compass-components/src/components/compass-components-provider.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import LeafyGreenProvider from '@leafygreen-ui/leafygreen-provider';
+import { ConfirmationModalArea } from '../hooks/use-confirmation';
+import { ToastArea } from '../hooks/use-toast';
+import { GuideCueProvider } from './guide-cue/guide-cue';
+import { SignalHooksProvider } from './signal-popover';
+
+type GuideCueProviderProps = React.ComponentProps<typeof GuideCueProvider>;
+
+type CompassComponentsProviderProps = {
+  /**
+   * Dark mode to be passed to the leafygreen provider. If not provided, the
+   * value will be derived from the system settings
+   */
+  darkMode?: boolean;
+  /**
+   * Leafygreen provider portalContainer property as a ref
+   */
+  portalContainerRef?: React.RefObject<HTMLElement>;
+  /**
+   * Leafygreen provider scrollContainer property as a ref
+   */
+  scrollContainerRef?: React.RefObject<HTMLElement>;
+  /**
+   * Either React children or a render callback that will get the darkMode
+   * property passed as function properties
+   */
+  children?:
+    | React.ReactChildren
+    | (({
+        darkMode,
+      }: {
+        darkMode: boolean;
+        portalContainerRef: React.Ref<HTMLElement>;
+        scrollContainerRef: React.Ref<HTMLElement>;
+      }) => React.ReactElement | null);
+} & {
+  onNextGuideGue?: GuideCueProviderProps['onNext'];
+  onNextGuideCueGroup?: GuideCueProviderProps['onNextGroup'];
+} & React.ComponentProps<typeof SignalHooksProvider>;
+
+const darkModeMediaQuery = (() => {
+  return typeof window !== 'undefined' && window.matchMedia
+    ? window.matchMedia('(prefers-color-scheme: dark)')
```

Comment:
This works well for the electron and web and allows to remove dependency on electron apis, even though we wouldn't really need it for web where I expect mms to be passing a `darkMode` prop that will switch this component to "controlled" mode